health-checker/src/main/java/io/stargate/health/metrics/HealthCheckerTagsProvider.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.health.metrics;
+
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.jersey2.server.JerseyTags;
+import io.micrometer.jersey2.server.JerseyTagsProvider;
+import io.stargate.core.metrics.StargateTenantExtractor;
+import io.stargate.core.metrics.api.Metrics;
+import org.glassfish.jersey.server.ContainerRequest;
+import org.glassfish.jersey.server.ContainerResponse;
+import org.glassfish.jersey.server.monitoring.RequestEvent;
+
+/**
+ * Tag provider for the Jersey HTTP request.
+ *
+ * @see RequestEvent
+ * @see io.micrometer.jersey2.server.DefaultJerseyTagsProvider
+ */
+public class HealthCheckerTagsProvider implements JerseyTagsProvider {
+
+  private final Metrics metrics;
+
+  private final String module;
+
+  public HealthCheckerTagsProvider(Metrics metrics, String module) {
+    this.metrics = metrics;
+    this.module = module;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Iterable<Tag> httpRequestTags(RequestEvent event) {
```

Comment:
Sorry, I'm not familiar with `micrometer` or `jersey`... Could you explain briefly what the call path to this method looks like? 

Is it used for our Web endpoints only?

Would it make sense to introduce a base module for our HTTP interfaces and keep the common setup/processing code there? It does not have to be an OSGi bundle. A shared plain jar (embedded in each API bundle) might work too.

Reply:
OK so it's app. like this:

- Jersey provides a `ApplicationEventListener` that has a single method `public RequestEventListener onRequest(RequestEvent requestEvent);`
- You can implement `ApplicationEventListener` and `RequestEventListener` to effectively have listeners on each HTTP requests (this is actually how the dropwizard metrics are also collected) 
- `micrometer-jersey2` provides implementation of both for the micrometer
- when creating the `micrometer-jersey2` event listener you need to provide:
   1. `MetricRegistry` to record metrics (this would be our newly created Prometheus registry)
   2. name of the metric you want to save this to, currently defined by us as `server.http.requests`
   3. `TagsProvider`, the implementation that extract tags that should be used when metric is recorded. The `micrometer-jersey2` provides default impl of this, with some common tags extracting like http method, status code, etc. But we here can plug-in our own stuff and change tags as we like.